contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -781,7 +790,12 @@ contract OCR2DRRegistry is
       Commitment memory commitment = s_requestCommitments[requestId];
 
       // Check that the message sender is the subscription owner
-      if (msg.sender != s_subscriptionConfigs[commitment.subscriptionId].owner) {
+      // Do we want to only allow the subscription owner to time out a request?
+      // Should the oracle admin also be able to timeout requests so that the owner doesn't have to do it manually if there is an outage?
```

Comment:
Fair point. If we want to have that ability then we also need to have a way to collect all requests that happened in a given time period (tooling feature maybe?). I wonder if it would make sense to also have a bigger hammer like timeoutAll() or something like that.

Reply:
Since we will always be able to use an archive node, we can filter through all past events to get expired requestIds.

The only reason we need `getRequestIdsToTimeOut` is that we cannot expect that all users will have access to an archive node, so they need a way of "searching" for pending requests in the mapping.